source/fundamentals/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1221 @@
+~~~~~~~~~~~~~~
+
+You can perform a map operation on a value of either type ``MqlMap`` or
+``MqlEntry`` using the methods described in this section.
+
+.. tip::
+   
+   You should represent data as a map if the data maps
+   arbitrary keys such as dates or item IDs to values.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `entries() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#entries()>`__
+     - :manual:`$objectToArray </reference/operator/aggregation/objectToArray/>`
+
+   * - `get() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#get(com.mongodb.client.model.mql.MqlString)>`__
+     - *No corresponding operator*
+
+   * - `getKey() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlEntry.html#getKey()>`__
+     - *No corresponding operator*
+
+   * - `getValue() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlEntry.html#getValue()>`__
+     - *No corresponding operator*
+
+   * - `has() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#has(com.mongodb.client.model.mql.MqlString)>`__
+     - *No corresponding operator*
+
+   * - `merge() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#merge(com.mongodb.client.model.mql.MqlMap)>`__
+     - *No corresponding operator*
+
+   * - `set() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#set(com.mongodb.client.model.mql.MqlString,T)>`__
+     - *No corresponding operator*
+
+   * - `setKey() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlEntry.html#setKey(com.mongodb.client.model.mql.MqlString)>`__
+     - *No corresponding operator*
+
+   * - `setValue() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlEntry.html#setValue(T)>`__
+     - *No corresponding operator*
+
+   * - `unset() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#unset(com.mongodb.client.model.mql.MqlString)>`__
+     - *No corresponding operator*
+
+Suppose you have a collection of warehouse data where each document
+represents an individual warehouse. Each document contains a field that
+is a map of the quantities for individual item IDs. You want to
+represent this map as an array to highlight the two lowest stocked items in
+each warehouse.
+
+The ``entries()`` method returns the map entries in the ``inventory``
+field as an array. The ``minN()`` method finds the two lowest valued array
+entries for each document. This example stores the result
+as the new ``lowInventory`` field using the ``project()`` method.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   var inventory = current().getMap("inventory");
+
+   asList(project(fields(
+       computed("lowInventory", inventory
+           .entries()
+           .minN(of(2)))
```

Comment:
I don't think that the minN will work here in a non-accidental way. While it's true that it will compare entry documents, this is unreliable if the entry's fields are "out of order" (for example, the key, which is the name of the item, gets evaluated first to determine the "smallest" value, but actually it's the value that should be compared). (And there's no real way to force order for fields.)

I don't think we have a sort operator, so I'm not sure this can work at all. If I had to do it, I would map entries into arrays that placed the inventory first, and the name of the item second (via `entries().map(e -> ofArray(e.getValue(), e.getKey())`), and then I think (but would need to verify the docs on this) that the order would be assured because the first item of the array would be guaranteed to be compared before the second. But this gets a bit convoluted.

The example might have to be changed to use something that takes a function (like `v->v.getValue()`). Maybe instead of warehouses with items, there are items with warehouses, and you do `currentItem.warehouses.entries().sum(v -> v.getValue())` to count up the item's total quantity across all warehouses it's stored in? Or something with a similar pattern.